### PR TITLE
fix(provenance): reject partial year-led timestamps (#1657)

### LINKED ISSUE #1657: follow-up(#1575): marginal provenance timestamp-format / quote edge cases (won't-fix from #1648)
## Summary

Track the remaining marginal timestamp-format / quote-normalization edge cases surfaced by `cursor[bot]` / `chatgpt-codex-connector[bot]` on PR #1648 (#1575 PR2 — claim-level provenance spans) that were deliberately **not** fixed in-loop to keep review churn bounded (see #1520 working agreement: median review threads per substantive PR < 15).

These are all P2 / P3 same-class micro-edge-cases in the `toStrictIsoTimestamp` / `locateQuoteOffsets` validators that the PR's 5 prior fix commits already substantially hardened. They have low realistic likelihood (provider/imported turn timestamps are full ISO-8601 strings from host clocks) and bounded consequence (no crash, no data loss, no security impact — at worst a fabricated-but-valid `observedAt` used only for citation ordering).

### 1. Partial timestamps admitted before normalization (chatgpt-codex-connector, PRRT_kwDORJXyws6OdKaD, P2)

`toStrictIsoTimestamp` in `packages/remnic-core/src/provenance.ts` admits a partial-but-`Date.parse`-able turn timestamp such as `2026-05` or `2026-05T10:00` because the separator guard (`/[-:T]/`) passes and none of the Y/M/D validators below catch a `< 3`-component date. `Date.parse` then fills in the missing components (May 1, zero seconds) and the write path persists a fabricated `observedAt`.

Repro (verified on Node):
```
"2026-05"       -> 2026-05-01T00:00:00.000Z   // fabricated day/time
"2026-05T10:00" -> 2026-05-01T15:00:00.000Z   // fabricated; ymd regex also misparses hour as day
```

Suggested fix: require a complete date-or-timestamp shape (either full `YYYY-MM-DD` plus optional `THH:MM[:SS]`, or the strict ISO form) before accepting the normalized round-trip — reject anything with fewer than 3 numeric calendar components.

### 2. (Placeholder for future same-class threads)

Additional marginal timestamp-format / quote-normalization edge cases surfaced on #1575 PR2/PR3 will be appended here rather than implemented as in-loop churn.

## Why not fixed inline

- Same validator class already hardened by 5 fix commits on #1648 (bare-year rejection, calendar-overflow `isValidCalendarDate`, non-YMD overflow, overflow-ts rejection, quote sanitization/stripping).
- All instances are P2/P3 with low realistic likelihood (full-ISO provider timestamps are the norm).
- Implementing an endless tail of micro-edge-cases is the exact failure mode the #1520 working agreement and the converge protocol exist to prevent.

## Acceptance

When addressed, the fix lands as ONE commit covering the whole class (not one commit per micro-case), with a characterization test seeded by the repro above.

### PR DESCRIPTION
**Pull request by joshuaswarren**
## Summary

Closes #1657 — the partial timestamp-format edge case deferred from #1648 (#1575 PR2 review, codex thread `PRRT_kwDORJXyws6OdKaD`).

`toStrictIsoTimestamp` admitted partial-but-`Date.parse`-able turn timestamps such as `"2026-05"` and `"2026-05T10:00"`:
- the separator guard (`/[-:T]/`) passed,
- the `ymd` calendar validator was skipped for `"2026-05"` (only 2 components), and **misparsed the hour as the day** for `"2026-05T10:00"` (`\D` matched `T`),
- `Date.parse` then filled in the missing components, so the write path persisted a **fabricated** `observedAt` (`2026-05-01…`).

Verified repro on Node:
```
"2026-05"       -> 2026-05-01T00:00:00.000Z   // fabricated day/time
"2026-05T10:00" -> 2026-05-01T15:00:00.000Z   // fabricated; ymd misparses hour as day
```

## Change

ONE commit covering the whole class (per #1657 acceptance):
- **Guard**: in `toStrictIsoTimestamp`, require a complete three-component calendar date (`YYYY<sep>MM<sep>DD` where `<sep>` is a non-digit, non-`T`, non-`:` date separator) for any year-led timestamp before accepting the normalized round-trip. Reject partial year-led shapes. Non-numeric-month shapes (`"Jan 15 2026"`) do not start with a 4-digit year and fall through to `Date.parse` unchanged.
- **Defense-in-depth**: tighten the `ymd` regex from `\D` to `[^0-9T:]` so the capture can no longer cross the date/time boundary and grab the hour as the day.

Bounded consequence (per the issue): no crash, no data loss, no security impact — at worst a fabricated-but-valid `observedAt` used only for citation ordering. Low realistic likelihood (provider/imported turn timestamps are full ISO-8601 from host clocks).

The deferred doctor edge from the #1657 comment (provenance-coverage `readArchivedMemories()` swallowing `SecureStoreLockedError`) is **already addressed** on main (`operator-toolkit.ts` uses `Promise.allSettled` + `provenancePartialTiers` to surface a locked store as a partial-read "warn", citing the exact codex threads) — landed in c03cef755 (#1682). No change needed there.

## Tests

Characterization tests in `provenance-extraction.test.ts` seeded by the issue repro:
- partial `YYYY-MM` → rejected, epoch fallback (prove-fail before fix)
- partial `YYYY-MM T HH:MM` → rejected, epoch fallback, no fabricated date (prove-fail before fix)
- complete `YYYY-MM-DD` (no time) → still normalizes to UTC midnight (non-regression)

## Verification

- `provenance-extraction.test.ts`: 45/45 (2 new tests prove-failed on unfixed code, pass after fix)
- All provenance suites (`provenance-extraction/frontmatter/surfaces/memory-provenance/orchestrator-source-attribution`): 94/94
- `test:entity-hardening`: 246/246
- `check-ratchets`: OK
- `preflight:quick`: OK

Chokepoints used: none — hardening of the existing `toStrictIsoTimestamp` validator in the provenance path; no new surface.

---

> [!NOTE]
> **Low Risk**
> Localized validation in provenance timestamp coercion; worst case is unverified facts with epoch fallback for citation ordering, not auth or data loss.
> 
> **Overview**
> Hardens **`toStrictIsoTimestamp`** so provenance extraction no longer persists **fabricated `observedAt`** values when turn timestamps are partial but still accepted by `Date.parse` (e.g. `"2026-05"` → May 1, `"2026-05T10:00"` with the hour misread as the day).
> 
> **`toStrictIsoTimestamp`** now trims input first, then for any year-led string requires a **complete three-part calendar date** via new **`parseYearLedDatePrefix`** (numeric or textual months, real date separators—not `T`/`:` or the space before a time). Incomplete, overflow, and boundary cases return `undefined` instead of normalizing; complete shapes like `YYYY-MM-DD` and full year-first month names still round-trip. The older separate `ymd` overflow check is folded into this path; M/D/Y overflow handling is unchanged.
> 
> **`provenance-extraction.test.ts`** adds regression coverage for partial numeric/textual year-led timestamps, whitespace bypass, space-separated partials, textual overflow, and non-regression for complete dates.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 2a7559c817709c89624c6085fed4c6f362d6696d. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/remnic-core/src/provenance.ts [hunk 1]
```diff
@@ -238,6 +244,50 @@ function isValidCalendarDate(y: number, mo: number, da: number): boolean {
...
+ * true calendar day — not the hour before a `:` time separator (the
+ * `(?![\d:])` lookahead rejects "2026-05 10:00" capturing "10" as the day).
+ * Returns the numeric components for a complete, valid date; `null` for an
+ * incomplete ("2026-05", "2026-Jan"), overflowed ("2026-02-30",
+ * "2026-Feb-30"), or non-year-led prefix. Centralizes the fabricated-date
+ * rejection for the toStrictIsoTimestamp normalization path across numeric
+ * and textual months in one rule.
+ */
+function parseYearLedDatePrefix(s: string): { y: number; mo: number; da: number } | null {
+  const m = /^(\d{4})[^0-9T](\d{1,2}|[A-Za-z]{3,})[^0-9T:](\d{1,2})(?![\d:])/.exec(s);
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/provenance.ts:282]: Deferring per the issue's own anti-tail guidance — this is the endless punctuation-variant tail, not the fabrication class #1657 targets. `Date.parse` returns the correct calendar day for these (no fabricated observedAt); the only effect is the source drops to the unverified epoch fallback (a missed citation). Tracked as a single follow-up (#1723) rather than chased in-loop, per the #1520 working agreement ("implementing an endless tail of micro-edge-cases is the exact failure mode the converge protocol exists to prevent"). Resolving this thread; the core fabrication fix and the unified numeric+textual-month guard landed in #1714.
